### PR TITLE
ixfrdist: Allow setting the inbound AXFR timeout

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
When transferring huge zones, we might not make in the hardcoded 10 seconds. This PR makes `AXFRRetriever::getChunk()` and `timeoutReadn()` get an optional timeout parameter and hooks it up to ixfrdist.

We'll nee dto hook this up to the recursor and the auth in the future as well.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/ixfrdist.cc [hunk 1]
```diff
@@ -707,6 +710,7 @@ int main(int argc, char** argv) {
       ("server-address", po::value<string>()->default_value("127.0.0.1:5300"), "server address")
       ("work-dir", po::value<string>()->default_value("."), "Directory for storing AXFR and IXFR data")
       ("keep", po::value<uint16_t>()->default_value(KEEP_DEFAULT), "Number of old zone versions to retain")
+      ("axfr-timeout", po::value<uint16_t>()->default_value(AXFRTIMEOUT_DEFAULT), "Timeout in seconds for an AXFR to complete")
```
Comment (rgacogne) [inline pdns/ixfrdist.cc]: It looks like it configures the timeout for getting a chunk of the `AXFR`, not for the whole `AXFR`, right?

Comment (pieterlexis) [reply pdns/ixfrdist.cc]: Correct, the wording is a bit off...

Comment (rgacogne) [reply pdns/ixfrdist.cc]: One option would be to keep track of the total elapsed time in `AXFRRetriever`, as we do for the number of bytes, but I'm not sure if that's you have in mind?

Comment (pieterlexis) [reply pdns/ixfrdist.cc]: That could also work, but that should be initialized on the first `getChunk` and be checked every consecutive call to `getChunk`. It also does not feel right.

**Pull request merged**